tools/bittool.cc

```diff
@@ -170,14 +170,28 @@ struct DeviceIdGetter {
 			            ArchType::ConfRegType::IDCODE);
 		    });
 		if (idcode_packet != reader->end()) {
-			if (idcode_packet->data().size() != 1) {
-				std::cerr
-				    << "Write to IDCODE with word_count != 1"
-				    << std::endl;
-				return 1;
+			if (std::is_same<ArchType, xilinx::Spartan6>::value) {
```

Comment:
This seems weird.  Can think of a way to make a better way to do this in bittool and bitread (which both have a similar block)?

Reply:
At the time when this was done, I tried to leave the most of the untouched and focused on the relevant part. 
However, this part, and a probably also some other of the original code could use some refactor.